crates/table/src/table.rs

```diff
@@ -439,7 +444,7 @@ impl Table {
 
         // SAFETY: We've checked above that `self.is_row_present(ptr)`.
         unsafe { self.delete_unchecked(blob_store, ptr) }
-
+        self.row_count -= 1;
```

Comment:
Similarly should the decrement happen inside `delete_internal`? I think replay might not go through this code path. It might call `delete_internal` directly.